### ISSUE TITLE
InvalidArgumentError using tf.learn and eval

### ISSUE THREAD

**Issue by dennybritz** [tensorflow/tensorflow#8701]
See https://github.com/google/seq2seq/issues/103 for details and user logs.

TLDR; I'm using tf.learn and for some people the evaluation fails with shape errors. This seems to be some kind of GPU memory sharing issue, as subsequent runs seem to consistently increase the shape size:

```
InvalidArgumentError (see above for traceback): logits and labels must have the same first dimension, got logits shape [1280,36240] and labels shape [6272]

InvalidArgumentError (see above for traceback): logits and labels must have the same first dimension, got logits shape [2304,36240] and labels shape [6272]

InvalidArgumentError (see above for traceback): logits and labels must have the same first dimension, got logits shape [4480,36240] and labels shape [6272]
```

Evaluation works independently when there is no training in progress. It also doesn't happen when using the CPU only.

I personally have run into similar issues before then multiple processes were trying to share the GPU, but that shouldn't be the case here.

**Comment by aselle** [comment]
@dennybritz, can you identify which line of python caused this error? My guess is this is related to the argument reorder on metrics... from the TensorFlow release notes.
```
Change arg order for {softmax,sparse_softmax,sigmoid}_cross_entropy_with_logits to be (labels, predictions), and force use of named args.
```
So go find those and insert explicit keyword argument labels appropriately and it should fix it. Let me know if that fixes you problem. Thanks

**Comment by dennybritz** [comment]
I don't think that is the issue. All arguments are named and it only happens to some users (I don't have this problem myself). Also, if you read the original issue you can see that some users managed to "fix" it by disabling bucketing or reducing the dev data.

It looks very much like some kind of GPU memory issue to me.

**Comment by aselle** [comment]
I'm sorry I missed that. I've seen many examples of shape errors being caused by that change, so it is worth a shot.

 Since this issue links to a incomplete summary and then 4 more issues which are themselves just blobs of erros, it's pretty hard to follow what is going on. Let me ask a few clarifying questions, since you've spent a good deal of time looking at this it appears.

1. What versions of TensorFlow are people using?  "compiled from source from the master branch" is not useful since master changes all the time.
2. Did all of them manage to run it on CPU only successfully?
3. What versions of CUDA are people using? Maybe there is a pattern.
4. What is the command I can run to try this example? Where is the code?

**Comment by dennybritz** [comment]
1. `1.0.0` or `1.0.1` - The code only works with that version.
2. The people I've talked to, yes. But I will try to ask more.
3. Not sure, will try to ask.
4. The command in this tutorial: https://google.github.io/seq2seq/nmt

**Comment by dennybritz** [comment]
So, this seems to be related to `train_and_evaluate` in tf.learn, see https://github.com/google/seq2seq/issues/103#issuecomment-293796457.

It seems like the only solution right now is to monkeypatch the Experiment class?

**Comment by dennybritz** [comment]
This has the patch: https://github.com/google/seq2seq/pull/173

**Comment by drpngx** [comment]
Did you mean to reopen?

**Comment by dennybritz** [comment]
Yes. I patched it in my code, but that doesn't seem right. I wonder what
the "official" solution is.

**Comment by drpngx** [comment]
OK, I wonder if it's best to use the seq2seq thread or this one. @sguada what do you think of this issue? It appears that it is related to `tf.learn`.

**Comment by itsmeolivia** [comment]
Automatically closing due to lack of recent activity. Since this issue is old at this point, please reopen the issue if it still occurs when tried with the latest version of Tensorflow. Thank you.

**Comment by jinyu121** [comment]
I meet the same problem.

TF versio: 1.3.1
python version: 3.5.2
code: https://gist.github.com/jinyu121/2c2e22f984a993a86f11f1be98f1d14c
data: the flower dataset used in tensorflow examples
data convert: create train and test TXT file with the format `image_path image_label`

--- 

log

```
INFO:tensorflow:Using default config.
INFO:tensorflow:Using config: {'_save_checkpoints_secs': 600, '_keep_checkpoint_every_n_hours': 10000, '_log_step_count_steps': 100, '_model_dir': '/tmp/cifar10_model', '_save_summary_steps': 100, '_save_checkpoints_steps': None, '_session_config': None, '_tf_random_seed': 1, '_keep_checkpoint_max': 5}
INFO:tensorflow:Create CheckpointSaverHook.
2017-10-03 15:54:51.821163: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2017-10-03 15:54:51.821198: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2017-10-03 15:54:51.821212: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2017-10-03 15:54:51.951386: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:893] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2017-10-03 15:54:51.951770: I tensorflow/core/common_runtime/gpu/gpu_device.cc:955] Found device 0 with properties: 
name: TITAN X (Pascal)
major: 6 minor: 1 memoryClockRate (GHz) 1.531
pciBusID 0000:01:00.0
Total memory: 11.90GiB
Free memory: 11.69GiB
2017-10-03 15:54:51.951790: I tensorflow/core/common_runtime/gpu/gpu_device.cc:976] DMA: 0 
2017-10-03 15:54:51.951798: I tensorflow/core/common_runtime/gpu/gpu_device.cc:986] 0:   Y 
2017-10-03 15:54:51.951809: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Creating TensorFlow device (/gpu:0) -> (device: 0, name: TITAN X (Pascal), pci bus id: 0000:01:00.0)
INFO:tensorflow:Restoring parameters from /tmp/cifar10_model/model.ckpt-1
INFO:tensorflow:Saving checkpoints for 2 into /tmp/cifar10_model/model.ckpt.
INFO:tensorflow:train_accuracy = 0.0234375, learning_rate = 0.1, cross_entropy = 2.27069
INFO:tensorflow:loss = 2.48257, step = 2
INFO:tensorflow:global_step/sec: 7.05856
INFO:tensorflow:train_accuracy = 0.316406, learning_rate = 0.1, cross_entropy = 0.947707 (14.168 sec)
INFO:tensorflow:loss = 1.16345, step = 102 (14.168 sec)
INFO:tensorflow:global_step/sec: 6.53794
INFO:tensorflow:train_accuracy = 0.486979, learning_rate = 0.1, cross_entropy = 0.500609 (15.295 sec)
INFO:tensorflow:loss = 0.714825, step = 202 (15.295 sec)
INFO:tensorflow:global_step/sec: 6.64687
INFO:tensorflow:train_accuracy = 0.558594, learning_rate = 0.1, cross_entropy = 0.557549 (15.045 sec)
INFO:tensorflow:loss = 0.7716, step = 302 (15.045 sec)
INFO:tensorflow:global_step/sec: 6.63171
INFO:tensorflow:train_accuracy = 0.621875, learning_rate = 0.1, cross_entropy = 0.359395 (15.079 sec)
INFO:tensorflow:loss = 0.577357, step = 402 (15.079 sec)
INFO:tensorflow:global_step/sec: 6.63793
INFO:tensorflow:train_accuracy = 0.670573, learning_rate = 0.1, cross_entropy = 0.329962 (15.065 sec)
INFO:tensorflow:loss = 0.554944, step = 502 (15.065 sec)
INFO:tensorflow:global_step/sec: 6.59523
INFO:tensorflow:train_accuracy = 0.712054, learning_rate = 0.1, cross_entropy = 0.123278 (15.163 sec)
INFO:tensorflow:loss = 0.3548, step = 602 (15.163 sec)
INFO:tensorflow:global_step/sec: 6.59328
INFO:tensorflow:train_accuracy = 0.748047, learning_rate = 0.1, cross_entropy = 0.0205212 (15.167 sec)
INFO:tensorflow:loss = 0.258983, step = 702 (15.167 sec)
INFO:tensorflow:global_step/sec: 6.62656
INFO:tensorflow:train_accuracy = 0.776042, learning_rate = 0.1, cross_entropy = 0.0149553 (15.091 sec)
INFO:tensorflow:loss = 0.248592, step = 802 (15.091 sec)
INFO:tensorflow:global_step/sec: 6.6181
INFO:tensorflow:train_accuracy = 0.792969, learning_rate = 0.1, cross_entropy = 0.128825 (15.110 sec)
INFO:tensorflow:loss = 0.361258, step = 902 (15.110 sec)
INFO:tensorflow:Saving checkpoints for 1001 into /tmp/cifar10_model/model.ckpt.
INFO:tensorflow:Loss for final step: 0.321334.
INFO:tensorflow:Starting evaluation at 2017-10-03-07:57:45
2017-10-03 15:57:45.654548: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Creating TensorFlow device (/gpu:0) -> (device: 0, name: TITAN X (Pascal), pci bus id: 0000:01:00.0)
INFO:tensorflow:Restoring parameters from /tmp/cifar10_model/model.ckpt-1001
2017-10-03 15:57:49.237303: W tensorflow/core/framework/op_kernel.cc:1192] Out of range: End of sequence         ##########HERE##########HRER##########
	 [[Node: IteratorGetNext = IteratorGetNext[output_shapes=[[?,32,32,3], [?,10]], output_types=[DT_FLOAT, DT_FLOAT], _device="/job:localhost/replica:0/task:0/cpu:0"](OneShotIterator)]]
INFO:tensorflow:Finished evaluation at 2017-10-03-07:57:49
INFO:tensorflow:Saving dict for global step 1001: accuracy = 0.821185, global_step = 1001, loss = 0.840976
{'accuracy': 0.82118529, 'loss': 0.8409763, 'global_step': 1001}
INFO:tensorflow:Create CheckpointSaverHook.
2017-10-03 15:57:52.058620: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Creating TensorFlow device (/gpu:0) -> (device: 0, name: TITAN X (Pascal), pci bus id: 0000:01:00.0)
INFO:tensorflow:Restoring parameters from /tmp/cifar10_model/model.ckpt-1001
INFO:tensorflow:Saving checkpoints for 1002 into /tmp/cifar10_model/model.ckpt.
INFO:tensorflow:train_accuracy = 0.984375, learning_rate = 0.1, cross_entropy = 0.0718204
INFO:tensorflow:loss = 0.309044, step = 1002
INFO:tensorflow:global_step/sec: 6.99782
INFO:tensorflow:train_accuracy = 0.988281, learning_rate = 0.1, cross_entropy = 0.0272659 (14.290 sec)
INFO:tensorflow:loss = 0.261741, step = 1102 (14.290 sec)
INFO:tensorflow:global_step/sec: 6.77749
INFO:tensorflow:train_accuracy = 0.992188, learning_rate = 0.1, cross_entropy = 0.00188445 (14.755 sec)
INFO:tensorflow:loss = 0.231876, step = 1202 (14.755 sec)
INFO:tensorflow:global_step/sec: 6.76658
INFO:tensorflow:train_accuracy = 0.994141, learning_rate = 0.1, cross_entropy = 0.00136539 (14.779 sec)
INFO:tensorflow:loss = 0.223648, step = 1302 (14.778 sec)
INFO:tensorflow:global_step/sec: 6.77486
INFO:tensorflow:train_accuracy = 0.995313, learning_rate = 0.1, cross_entropy = 0.000284866 (14.760 sec)
INFO:tensorflow:loss = 0.213993, step = 1402 (14.760 sec)
INFO:tensorflow:global_step/sec: 6.7589
INFO:tensorflow:train_accuracy = 0.996094, learning_rate = 0.1, cross_entropy = 0.000372966 (14.795 sec)
INFO:tensorflow:loss = 0.205791, step = 1502 (14.795 sec)
INFO:tensorflow:global_step/sec: 6.73266
INFO:tensorflow:train_accuracy = 0.996652, learning_rate = 0.1, cross_entropy = 0.000321502 (14.853 sec)
INFO:tensorflow:loss = 0.197838, step = 1602 (14.853 sec)
INFO:tensorflow:global_step/sec: 6.73123
INFO:tensorflow:train_accuracy = 0.99707, learning_rate = 0.1, cross_entropy = 0.000238253 (14.856 sec)
INFO:tensorflow:loss = 0.190108, step = 1702 (14.856 sec)
INFO:tensorflow:global_step/sec: 6.76846
INFO:tensorflow:train_accuracy = 0.997396, learning_rate = 0.1, cross_entropy = 0.000165852 (14.775 sec)
INFO:tensorflow:loss = 0.182676, step = 1802 (14.775 sec)
INFO:tensorflow:global_step/sec: 6.74329
INFO:tensorflow:train_accuracy = 0.997656, learning_rate = 0.1, cross_entropy = 0.000469119 (14.829 sec)
INFO:tensorflow:loss = 0.175907, step = 1902 (14.830 sec)
INFO:tensorflow:Saving checkpoints for 2001 into /tmp/cifar10_model/model.ckpt.
INFO:tensorflow:Loss for final step: 0.169032.
INFO:tensorflow:Starting evaluation at 2017-10-03-08:00:38
2017-10-03 16:00:39.299323: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Creating TensorFlow device (/gpu:0) -> (device: 0, name: TITAN X (Pascal), pci bus id: 0000:01:00.0)
INFO:tensorflow:Restoring parameters from /tmp/cifar10_model/model.ckpt-2001
2017-10-03 16:00:42.905022: W tensorflow/core/framework/op_kernel.cc:1192] Out of range: End of sequence         ##########HERE##########HRER##########
	 [[Node: IteratorGetNext = IteratorGetNext[output_shapes=[[?,32,32,3], [?,10]], output_types=[DT_FLOAT, DT_FLOAT], _device="/job:localhost/replica:0/task:0/cpu:0"](OneShotIterator)]]
INFO:tensorflow:Finished evaluation at 2017-10-03-08:00:42
INFO:tensorflow:Saving dict for global step 2001: accuracy = 0.384196, global_step = 2001, loss = 3.59174
{'accuracy': 0.38419619, 'loss': 3.5917444, 'global_step': 2001}
INFO:tensorflow:Create CheckpointSaverHook.
2017-10-03 16:00:45.340488: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Creating TensorFlow device (/gpu:0) -> (device: 0, name: TITAN X (Pascal), pci bus id: 0000:01:00.0)
```
